### PR TITLE
#44 label_encoder

### PR DESCRIPTION
**Pull request by abhinavmalhotra01**
implemented label encoder function

### CODE REVIEW

File: src/slowmokit/methods/preprocessing/label_encoder.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
+ * @file methods/preprocessing/label_encoder.hpp
+ *
+ * Implementation of label encoder
+ */
+
+#include "label_encoder.hpp"
+
+template<class T>
+std::vector<int> label_encoder(std::vector<T>& input ){
```
Comment (ken1000minus7) [inline src/slowmokit/methods/preprocessing/label_encoder.cpp:10]: Variables and function names should be in camel case

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]

**Review by harshjohar — APPROVED** [review]
lgtm (only file structure)

File: src/slowmokit/methods/preprocessing/label_encoder.cpp (hunk 1 above)
Comment (abhinavmalhotra01) [reply src/slowmokit/methods/preprocessing/label_encoder.cpp:10]: oh , changing them

Comment (abhinavmalhotra01) [reply src/slowmokit/methods/preprocessing/label_encoder.cpp]: changed sir

**Comment by harshjohar** [comment]
Screenshot 2023-02-07 at 1 01 52 AM

fix this

**Comment by abhinavmalhotra01** [comment]
> Screenshot 2023-02-07 at 1 01 52 AM
> 
> fix this

fixed

**Review by uttammittal02 — CHANGES_REQUESTED** [review]
Add an example file [here](https://github.com/PEC-CSS/slowmokit/tree/main/examples/preprocessing)

**Comment by ken1000minus7** [comment]
Pull fresh changes from the main to resolve the conflicts

**Comment by abhinavmalhotra01** [comment]
> Add an example file [here](https://github.com/PEC-CSS/slowmokit/tree/main/examples/preprocessing)

added

**Comment by abhinavmalhotra01** [comment]
> Pull fresh changes from the main to resolve the conflicts

merged

File: CMakeLists.txt [hunk 2]
```diff
@@ -40,10 +40,12 @@ add_library(slowmokit
...
         src/slowmokit/methods/neighbors/bernoulli_nb/bernoulli_nb.cpp
+        src/slowmokit/methods/preprocessing/label_encoder.cpp
+        src/slowmokit/methods/preprocessing/label_encoder.hpp
         src/slowmokit/methods/metrics/classification_report.hpp
         src/slowmokit/methods/metrics/classification_report.cpp
         src/slowmokit/methods/neighbors/gaussian_nb.hpp
         src/slowmokit/methods/neighbors/gaussian_nb/gaussian_nb.cpp
         src/slowmokit/methods/neighbors/gaussian_nb/gaussian_nb.hpp
         src/slowmokit/methods/preprocessing/one_hot_encoder.hpp
-        src/slowmokit/methods/preprocessing/one_hot_encoder.cpp)
```
Comment (ken1000minus7) [inline CMakeLists.txt:49]: Don't remove `)`

File: CMakeLists.txt [hunk 3]
```diff
@@ -40,10 +40,12 @@ add_library(slowmokit
...
+        src/slowmokit/methods/preprocessing/one_hot_encoder.cpp
```
Comment (uttammittal02) [inline CMakeLists.txt]: A Closing bracket is needed at the end of this line

File: CMakeLists.txt (hunk 2 above)
Comment (abhinavmalhotra01) [reply CMakeLists.txt]: yes sir, changed it 😢

**Review by uttammittal02 — CHANGES_REQUESTED** [review]

**Review by ken1000minus7 — APPROVED** [review]
Lgtm

**Review by uttammittal02 — APPROVED** [review]
Lgtm

**Pull request merged**